src/test/java/com/google/research/bleth/services/DatabaseServiceTest.java

```diff
@@ -0,0 +1,247 @@
+package com.google.research.bleth.services;
+
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.dev.LocalDatastoreService;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.research.bleth.simulator.Agent;
+import com.google.research.bleth.simulator.Board;
+import com.google.research.bleth.simulator.Location;
+import junit.framework.TestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class DatabaseServiceTest extends TestCase {
```

Comment:
I got lost in the scenarios here... Are these complex scenarios really necessary?

Why do you need more than
* Write a specific (non-empty) real board, verify it reads back correctly
* Same for estimated
* Verify deletion works

Do we really care about these "multiple simulation ID" cases? Can they ever happen in practice?

Reply:
I agree we may need more than just the 3 simple scenarios I listed, but - you still have far too many IMO.
For example, there's no real need to verify every combination of parameters - a single testcase can verify that changing simulation ID is enough to get a different board; another testcase verifies that changing just the round number also gets a different board; and that's it.